odoo/resource/account_move/invoice.py

```
@@ -0,0 +1,180 @@
+from odoo.resource.product_product.spec import ProductData
+from odoo.resource.res_partner.spec import PartnerData, PartnerType
+
+logger = logging.getLogger(__name__)
+
+
+class OdooInvoiceResource:
+    def get_charge_item_base_price(self, charge_item: ChargeItem):
+        for item in charge_item.unit_price_components:
+            if item["monetary_component_type"] == MonetaryComponentType.base.value:
+                return item["amount"]
+        raise Exception("Base price not found")
```

Ruff (0.14.3)

Create your own exception

(TRY002)